org.adempiere.base/src/org/compiere/model/MRole.java

```diff
@@ -3344,4 +3344,21 @@ private boolean isAccessLevelCompatible(String accessLevel) {
 		return access;
 	}
 
+	/**
+	 * Does the table is excluded for current role (this method doesn't check the level of the table, use isTableAccess for this purpose)
+	 * @param tableID ID of the table
+	 * @return
+	 */
+	public boolean isTableExcluded(int tableID)
+	{
+		loadTableAccess(false);
+
+		for (int i = 0; i < m_tableAccess.length; i++) {
+			if (m_tableAccess[i].isExclude() && m_tableAccess[i].getAD_Table_ID() == tableID)
```

Comment:
Hi @nmicoud - I tested the patch successfully, however, because the table access can be defined for many other things is better to rename this method to isTableAccessExcluded, and the validation must verify access type and readonly, like suggested here:
[mod257.txt](https://github.com/idempiere/idempiere/files/5251702/mod257.txt)